### ISSUE TITLE
iOS support

### ISSUE THREAD

**Issue by JohnPaulHarold** [alemangui/pizzicato#18]
I tried the demo site on an iPhone (recent one, not sure the exact model), and the demos didn't work. While I've not gone through and done exhaustive tests to work out why, I'm fairly confident it's to do with Pizzicato initialising itself and therefore an AudioContext as soon as the script is processed. 

For iOS, the context needs to be initialised inside a user invoked method, such as a touchend/click event.

**Comment by alemangui** [comment]
Interesting, I'll take a look. thanks!

**Comment by sksar** [comment]
I tried it too, but on iOS 8 it works just fine...

**Comment by alemangui** [comment]
I did some tests with iOS 9.3.4 and it seems to work well. Have you checked you're not in silent/vibration mode?

While at it, I modified the media queries so the site becomes usable with small screens. It should be less painful now.

**Comment by JohnPaulHarold** [comment]
@alemangui I tested it some more over the weekend as well. It's weirder than I thought. Some effects I couldn't get to work at all, such as flanger, and others did, such as RingMod. This latest testing was on an iPad2, running whatever the latest, or semi latest 9.x.x release is. I'll do some more testing and see if there's something afoot.

As for the site, I was about to raise a ticket on that as well. Glad you're on top of it.

**Comment by lukasIO** [comment]
hey, are there any news on ios support? on ios 10 i cannot play any of the demos :\

**Comment by alemangui** [comment]
I tried again but on my side it works well. There are two things that may get in the way though...

1- Slow loading time in Pizzicato's site. The audio files are pretty big, and loading them from mobile can be painful. This may give the impression of Pizzicato not working, when it's the files that are still loading.

2- Sounds from 'input' use the ```getUserMedia``` function - which is not supported at all in iOS. All the other demos should work though.

**Comment by JohnPaulHarold** [comment]
I'll just reiterate, I raised this ticket as all my previous dealings with WebAudio and iOS required that the context be created within a user invoked event. So for me, I wired in a click event and then created the context inside the event callback. However, after I raised this, others said they didn't see this, and that the demo worked fine on iOS. I'm not 100% sure why or how Pz is immune to this, or if iOS has changed stance. It's very confusing, as I've also seen iOS devices work, and others not work. However, I've not really been involved with WebAudio much lately so I've not really given the matter much further investigation.
